src/Game/Jobs/DRG.ts

```diff
@@ -0,0 +1,1069 @@
+		}
+
+		let combo_value = -1;
+		// chaos combo
+		if (CHAOS_COMBO_MAP.has(skillName)) {
+			this.tryConsumeResource("DRG_HEAVENS_COMBO_TRACKER", true);
+			resType = "DRG_CHAOS_COMBO_TRACKER";
+			combo_value = CHAOS_COMBO_MAP.get(skillName)
+				? Number(CHAOS_COMBO_MAP.get(skillName))
+				: -1;
+		}
+		// heavens combo
+		else if (HEAVENS_COMBO_MAP.has(skillName)) {
+			this.tryConsumeResource("DRG_CHAOS_COMBO_TRACKER", true);
+			resType = "DRG_HEAVENS_COMBO_TRACKER";
+			combo_value = HEAVENS_COMBO_MAP.get(skillName)
+				? Number(HEAVENS_COMBO_MAP.get(skillName))
+				: -1;
+		}
+		// drakes bane
+		else if (skillName === "DRAKESBANE") {
+			this.tryConsumeResource("DRG_HEAVENS_COMBO_TRACKER", true);
+			this.tryConsumeResource("DRG_CHAOS_COMBO_TRACKER", true);
+			return;
+		}
+
+		if (this.resources.get(resType).availableAmount() === combo_value) {
+			this.resources.get(resType).gain(1);
+			this.enqueueResourceDrop(resType);
+		} else {
+			this.tryConsumeResource(resType, true);
+		}
+	}
+
+	// Return true if the active combo buff is up, or meikyo is active.
+	// Does not advance the combo state.
+	checkCombo(requiredCombo: DRGResourceKey): boolean {
+		return this.hasResourceAvailable(requiredCombo);
+	}
+
+	// gain a scale
+	gainScale(scales: number) {
+		if (this.resources.get("FIRSTMINDS_FOCUS").availableAmount() + scales > 2) {
+			controller.reportWarning(WarningType.ScaleOvercap);
+		}
+		this.resources.get("FIRSTMINDS_FOCUS").gain(scales);
+	}
+
+	refreshBuff(rscType: ResourceKey, delay: number) {
+		// buffs are applied on hit, so apply it after a delay
+		this.addEvent(
+			new Event("gain buff", delay, () => {
+				this.resources.get(rscType).gain(1);
+				this.enqueueResourceDrop(rscType);
+			}),
+		);
+	}
+}
+
+// === SKILLS ===
+// Abilities will display on the hotbar in the order they are declared here. If an ability has an
+// `autoDowngrade` (i.e. it replaces a previous ability on the hotbar), it will not have its own
+// slot and instead take the place of the downgrade ability.
+//
+// If an ability appears on the hotbar only when replacing another ability, it should have
+// `startOnHotbar` set to false, and `replaceIf` set appropriately on the abilities to replace.
+
+const makeWeaponskill_DRG = (
+	name: ActionKey,
+	unlockLevel: number,
+	params: {
+		autoUpgrade?: SkillAutoReplace;
+		autoDowngrade?: SkillAutoReplace;
+		assetPath?: string;
+		replaceIf?: ConditionalSkillReplace<DRGState>[];
+		startOnHotbar?: boolean;
+		potency?: number | Array<[TraitKey, number]>;
+		falloff?: number;
+		recastTime?: number | ResourceCalculationFn<DRGState>;
+		combo?: {
+			potency: number | Array<[TraitKey, number]>;
+			resource: DRGResourceKey;
+			resourceValue: number;
+		};
+		positional?: {
+			potency: number | Array<[TraitKey, number]>;
+			comboPotency: number | Array<[TraitKey, number]>;
+			location: "flank" | "rear";
+		};
+		jobPotencyModifiers?: PotencyModifierFn<DRGState>;
+		applicationDelay?: number;
```

Comment:
You should make `applicationDelay` mandatory to ensure you don't forget to set it for a skill.

```suggestion
		applicationDelay: number;
```